HHS/Head-Start-TTADP
# [TTAHUB-2644] Add goal name filter to grant scopes (for Recipient search)

## Description of change
This change just puts the backend code in place for the goal name filter. There will be several of these PRs incoming; they are all prerequisite for the frontend ticket, which will come last.

## How to test
Evaluate the code. You can attach "logging: console.log" to the options passed to the Recipient.findAlls added to the test to see the SQL generated if that helps evaluate the correctness. Ensure the test adequately covers the use case.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2644


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
